frontend/src/pages/user/shop/ShopSearch.tsx

```diff
@@ -0,0 +1,569 @@
+import '@components/style.css';
+import '@style/global.css';
+
+import { Col, Form, Row, Offcanvas } from 'react-bootstrap';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faFilter } from '@fortawesome/free-solid-svg-icons';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { useState, useEffect } from 'react';
+import { useMutation } from '@tanstack/react-query';
+
+import GoodsItem from '@components/GoodsItem';
+import TButton from '@components/TButton';
+import { getShopSearchUrl } from '@components/SearchBar';
+
+const PhoneOffCanvasStyle = {
+  width: '330px',
+  border: ' 1px solid var(--button_light, #60998B)',
+  background: ' var(--bg, #061E18)',
+  boxShadow: '0px 4px 20px 2px #60998B',
+  padding: '4% 10% 10% 10%',
+  color: 'white',
+};
+
+export interface FilterProps {
+  maxPrice: number | null;
+  minPrice: number | null;
+  maxStock: number | null;
+  minStock: number | null;
+  haveCoupon: boolean | null;
+  sortBy: 'price' | 'stock' | 'sales' | 'relevancy' | null;
+  order: 'asc' | 'desc' | null;
+}
+
+export interface SearchProps extends FilterProps {
+  q: string;
+}
+
+interface ProductProps {
+  id: number;
+  image_url: string;
+  name: string;
+  price: string;
+  stock: number;
+}
+
+// eslint-disable-next-line react-refresh/only-export-components
+export const defaultFilterValues: FilterProps = {
```

Comment:
why ignore this

Reply:
because it will give you `Fast refresh only works when a file only export components. Use a new file to share constant or functions between` and i don't want to create a new file for this

Reply:
why not

Reply:
I mean having a component that exports more than just components seems weird doesn't it